courses/management/commands/create_courseware.py

```diff
@@ -0,0 +1,191 @@
+            type=str,
+            help="The readable ID for the courseware object. (For course runs, do not include the run tag.)",
+        )
+
+        parser.add_argument("title", type=str, help="The title of the object.")
+
+        parser.add_argument(
+            "--live",
+            action="store_true",
+            help="Make the object live (defaults to not).",
+        )
+
+        parser.add_argument(
+            "--self-paced",
+            action="store_true",
+            help="(Course run only) Make the course run self-paced.",
+        )
+
+        parser.add_argument(
+            "--create-run",
+            "--run-tag",
+            nargs="?",
+            type=str,
+            help="(Course and course run only) Create a run with the specified tag.",
+            metavar="create_run",
+        )
+
+        parser.add_argument(
+            "--run-url",
+            type=str,
+            nargs="?",
+            help="(Course and course run only) Create a run with the specified URL path.",
+        )
+
+        parser.add_argument(
+            "--program",
+            type=str,
+            nargs="?",
+            help="(Course only) Add the course to the specified program (readable ID or numeric ID).",
+        )
+
+        parser.add_argument(
+            "--program-position",
+            type=str,
+            nargs="?",
+            help="(Course only) The position the course should be in the program.",
+        )
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        if kwargs["type"] == "program":
+            if Program.objects.filter(readable_id=kwargs["courseware_id"]).exists():
+                self.stderr.write(
+                    self.style.ERROR(
+                        f"Program with ID {kwargs['courseware_id']} already exists."
+                    )
+                )
+                exit(-1)
+
+            new_program = Program.objects.create(
+                readable_id=kwargs["courseware_id"],
+                title=kwargs["title"],
+                live=kwargs["live"],
+            )
+
+            self.stdout.write(
+                self.style.SUCCESS(
+                    f"Created program {new_program.id}: {new_program.title} ({new_program.readable_id})."
+                )
+            )
+        elif kwargs["type"] == "course":
+            if Course.objects.filter(readable_id=kwargs["courseware_id"]).exists():
+                self.stderr.write(
+                    self.style.ERROR(
+                        f"Course with ID {kwargs['courseware_id']} already exists."
+                    )
+                )
+                exit(-1)
+
+            program = None
+
+            if "program" in kwargs and kwargs["program"] is not None:
+                try:
+                    program = Program.objects.filter(pk=kwargs["program"])
+                except:
+                    program = Program.objects.filter(readable_id=kwargs["program"])
```

Comment:
Should this be `get` instead of filter?  I was able to create a Course and specified a program value that didn't exist.  The course was created with a value in "position in program" but no "program" value.

Reply:
no, the indention on this got munged up.. the `first` call was supposed to be outside the if. If you specify a program that doesn't exist, it should revert to None, though.